src/gsy_e/models/area/scm_manager.py

```diff
@@ -196,81 +304,122 @@ def calculate_home_energy_bills(
         taxes_surcharges = home_data.taxes_surcharges
         feed_in_tariff = home_data.feed_in_tariff
 
+        slots_per_month = (duration(days=1) / GlobalConfig.slot_length) * monthrange(
+            self._time_slot.year, self._time_slot.month)[1]
+        marketplace_fee = home_data.marketplace_monthly_fee / slots_per_month
+        fixed_fee = home_data.fixed_monthly_fee / slots_per_month
+
         market_maker_rate_decreased_fees = (
                 market_maker_rate + grid_fees * (1.0 - self._grid_fees_reduction) +
                 taxes_surcharges)
         market_maker_rate_normal_fees = market_maker_rate + grid_fees + taxes_surcharges
 
+        home_bill = AreaEnergyBills(
+            marketplace_fee=marketplace_fee, fixed_fee=fixed_fee,
+            gsy_energy_bill=marketplace_fee + fixed_fee)
+
         base_energy_bill = (
-                home_data.energy_need_kWh * market_maker_rate_normal_fees -
-                home_data.energy_surplus_kWh * feed_in_tariff)
+                home_data.energy_need_kWh * market_maker_rate_normal_fees + marketplace_fee +
+                fixed_fee - home_data.energy_surplus_kWh * feed_in_tariff)
+        home_bill.base_energy_bill = base_energy_bill
 
         if home_data.allocated_community_energy_kWh > home_data.energy_need_kWh:
             if home_data.energy_surplus_kWh > 0.0:
-                gsy_energy_bill = -(
-                    home_data.energy_bought_from_community_kWh * market_maker_rate_decreased_fees +
-                    self.community_data.energy_sold_to_grid_kWh * feed_in_tariff)
+                home_bill.set_bought_from_community(
```

Comment:
I know it's not ready yet, but I cannot understand the concept of calculating the gsy bill for houses which have surplus and sell energy to community. Here in the loop I don't think so that we will ever have `energy_bought_from_community_kWh`, because house has energy surplus and was allocated with more energy than it needed. The another thing is that I do not see the method like `set_sold_to_community` here, which I suppose should be introduced to pay this house for it's contribution to community, and not only what it gave to the grid. But I am not sure if I get it right in the first place, so help me here.

Reply:
> Here in the loop I don't think so that we will ever have energy_bought_from_community_kWh, because house has energy surplus and was allocated with more energy than it needed.

This is not true actually. In the community there will be homes for which the supply is not enough and also homes that might have no supply at all (homes without PVs). These homes will still benefit by participating in the community by buying energy with decreased grid fees. 

> The another thing is that I do not see the method like set_sold_to_community here, which I suppose should be introduced to pay this house for it's contribution to community, and not only what it gave to the grid.

Admittedly the set_sold_to_community method is not there, the reason being because it is not used in the example calculations that we had from the research team https://docs.google.com/spreadsheets/d/1CS3jFzxv_QLw6aXr-TvwLqoe9LKqoUZb5ckiECHs0kk/edit#gid=0 . And it makes sense, because there will be no revenue at all from the "sold to community" trades for the producers.

Reply:
> > Here in the loop I don't think so that we will ever have energy_bought_from_community_kWh, because house has energy surplus and was allocated with more energy than it needed.
> 
> This is not true actually. In the community there will be homes for which the supply is not enough and also homes that might have no supply at all (homes without PVs). These homes will still benefit by participating in the community by buying energy with decreased grid fees.
> 
> > The another thing is that I do not see the method like set_sold_to_community here, which I suppose should be introduced to pay this house for it's contribution to community, and not only what it gave to the grid.
> 
> Admittedly the set_sold_to_community method is not there, the reason being because it is not used in the example calculations that we had from the research team https://docs.google.com/spreadsheets/d/1CS3jFzxv_QLw6aXr-TvwLqoe9LKqoUZb5ckiECHs0kk/edit#gid=0 . And it makes sense, because there will be no revenue at all from the "sold to community" trades for the producers, as far as I understood from the regulation at least. The only benefit of the community is in the form of decreased grid fees.

Reply:
Thanks. I am aware that there will be different homes, but I was referring to the scenario where `allocated_community_energy_kWh` > `energy_need_kWh` and `energy_surplus_kWh` > 0.0 , so the homes with no supply or with not enough supply won't be evaluated in this if statement. And that's why I think that there is no need for `set_bought_from_community`. The second issue, I found in the [doc](https://docs.google.com/document/d/1TeykhSzA952U8BtqvpqbVfpH5-_dzIQhCKxSykBv5G4/edit#) that the formula for the `GSy energy bill = - (total energy sold to the community * Community price + total energy sold to the grid * grid price) + PENALTY`. Which in my opinion makes sense. And according to spreadsheet, there is **GSy energy bill**  `R15`, which is calculated with use of `I21` and `J21` (`-(($I$21-I18)*(K18+M18*O18)+$J$21*L18)` it works only because there is only one house who is selling to the community, probably with more such houses, the formula should be revisit). So AFAIU the bill considers revenue for the energy sold to the community. But of course I could be totally wrong.

Reply:
> but I was referring to the scenario where allocated_community_energy_kWh > energy_need_kWh and energy_surplus_kWh > 0.0 , so the homes with no supply or with not enough supply won't be evaluated in this if statement. And that's why I think that there is no need for set_bought_from_community.

This case is not exactly true. If a house has energy_surplus > 0, it is guaranteed that its energy_need is 0. This is justified because the energy_need does not include self-consumed energy, meaning that each house first covers its own consumption with its own production and then evaluates whether it has a need or a surplus. The energy assignment to each house is independent from this mechanism, therefore it really needs to be there no matter if a house has energy need or energy surplus. 

> And according to spreadsheet, there is GSy energy bill R15, which is calculated with use of I21 and J21 (-(($I$21-I18)*(K18+M18*O18)+$J$21*L18) it works only because there is only one house who is selling to the community, probably with more such houses, the formula should be revisit). So AFAIU the bill considers revenue for the energy sold to the community. 

Here you do have a point though as far as I see. This formula does not make sense IMO too (because as you said, I21 is the total energy bought from the community , therefore I21 - I18 is the energy bought from the community from every house other than this one. Therefore for multiple houses, it will pay them more than the energy that they have sold to the grid). Same goes for the energy excess sold to the grid, with this formula each house is paid for the whole community excess (J21). I think that the best solution here is to change the formula to this `(-(I18*(K18+M18*O18) + J18*L18)`. If you agree I will reflect this change to the code too.

Reply:
I think that I get it now, thank you for your clarification. Yes, I agree to change the formula. Nevertheless I still miss the `total energy sold to the community * Community price` in our calculations, something like `sold_to_community` will work IMO, don't we need it ?

Reply:
This is true, that requirements are not clear here. I only see that energy sold to the community is in this [doc](https://docs.google.com/document/d/1TeykhSzA952U8BtqvpqbVfpH5-_dzIQhCKxSykBv5G4/edit#), but you are right, there is no algorithm distinguishing from grid vs community energy. Your proposition looks great for me, and is probably fair for the parties. Sorry for dragging this discussion, but I think that we are on the same page now. Thank you!

Reply:
Great to hear that we are in sync now ! I adapted the code to support what was discussed here